src/services/youtube.ts

```diff
@@ -48,46 +52,72 @@ export interface VideoIds {
 
 export class YouTubeService extends BaseClient {
   /**
-   * Fetches a transcript for a YouTube video.
+   * Handles YouTube Transcript operations.
    *
+   * @property transcript - Fetches a transcript for a YouTube video.
    * @param params - Parameters for fetching the transcript
-   * @param params.videoId - The YouTube video ID (provide either this OR url)
-   * @param params.url - The YouTube video URL (provide either this OR videoId)
-   * @param params.lang - Optional language code for the transcript
-   * @param params.text - Optional flag to return plain text instead of timestamped list
-   * @returns A promise that resolves to the video transcript
-   */
-  async transcript(params: TranscriptParams): Promise<Transcript> {
-    return this.fetch<Transcript>('/youtube/transcript', params);
-  }
-
-  /**
-   * Translates a YouTube video transcript to a specified language.
    *
+   * @property translate - Translates a YouTube video transcript.
    * @param params - Parameters for translating the transcript
-   * @param params.videoId - The YouTube video ID (provide either this OR url)
-   * @param params.url - The YouTube video URL (provide either this OR videoId)
-   * @param params.lang - The target language code for translation
-   * @param params.text - Optional flag to return plain text instead of timestamped list
-   * @returns A promise that resolves to the translated transcript
+   *
+   * @property batch - Batch fetches transcripts for multiple YouTube videos.
+   * @param params - Parameters for the transcript batch job
    */
-  async translate(params: TranslateParams): Promise<TranslatedTranscript> {
-    return this.fetch<TranslatedTranscript>(
-      '/youtube/transcript/translate',
-      params
-    );
-  }
+  transcript = Object.assign(
+    /**
+     * Fetches a transcript for a YouTube video.
+     */
+    async (params: TranscriptParams): Promise<Transcript> => {
+      return this.fetch<Transcript>('/youtube/transcript', params);
+    },
+    {
+      /**
+       * Batch fetches transcripts for multiple YouTube videos.
+       */
+      batch: async (
+        params: YoutubeTranscriptBatchRequest
+      ): Promise<YoutubeBatchJob> => {
+        this.validateBatchLimit(params);
+        return this.fetch<YoutubeBatchJob>(
+          '/youtube/transcript/batch',
+          params,
+          'POST'
+        );
+      },
+    }
+  );
```

Comment:
the JSDoc params are incorrectly assigned to whole object `transcript`, whereas some should be assigned to `batch` function